src/amuse/community/crater/interface.py

```diff
@@ -0,0 +1,577 @@
+        function.addParameter('Lgault', dtype='float64', direction=function.OUT)
+
+        function.result_type = 'int32'
+        function.can_handle_array = True
+        return function
+        
+
+class Crater(InCodeComponentImplementation):
+
+    def __init__(self, **options):
+        InCodeComponentImplementation.__init__(self,  CraterInterface(**options), **options)
+
+    def initialize_code(self):
+        print "ahhh."
+
+    def define_parameters(self, handler):
+        handler.add_method_parameter(
+            "get_target_density",
+            "set_target_density",
+            "target_density",
+            "initialize the density of the target",
+            default_value = 1.0 | units.kg/units.m**3
+            )
+        handler.add_method_parameter(
+            "get_target_type",
+            "set_target_type",
+            "target_type",
+            "initialize the type of target",
+            default_value = 1
+            )
+        handler.add_method_parameter(
+            "get_target_gravity",
+            "set_target_gravity",
+            "target_gravity",
+            "initialize the gravity of the target",
+            default_value = 9.8 | units.m/units.s**2
+            )
+        handler.add_method_parameter(
+            "get_projectile_density",
+            "set_projectile_density",
+            "projectile_density",
+            "initialize the density of the target",
+            default_value = 1
+            )
+        handler.add_method_parameter(
+            "get_projectile_diameter",
+            "set_projectile_diameter",
+            "projectile_diameter",
+            "initialize the diameter of the target",
+            default_value = 1
+            )
+        handler.add_method_parameter(
+            "get_impact_angle",
+            "set_impact_angle",
+            "impact_angle",
+            "initialize the impact angle",
+            default_value = 1
+            )
+        handler.add_method_parameter(
+            "get_impact_velocity",
+            "set_impact_velocity",
+            "impact_velocity",
+            "initialize the impact velocity",
+            default_value = 1
+            )
+        handler.add_method_parameter(
+            "get_crater_diameter",
+            "set_crater_diameter",
+            "crater_diameter",
+            "initialize the crater diameter",
+            default_value = 1
+            )
+        handler.add_method_parameter(
+            "get_crater_type",
+            "set_crater_type",
+            "crater_type",
+            "initialize the crater type",
+            default_value = 1
+            )
+        handler.add_method_parameter(
+            "get_formation_time",
+            "set_formation_time",
+            "formation_time",
+            "initialize the formation time",
+            default_value = 1
+            )
+        
+    def define_methods(self, handler):
```

Comment:
for next time: units can be specified in the @legacy_function definitions above (just add unit=... keyword to an argument). Then these definitions  (the define_methods as a whole) can be omitted...